src/webob/acceptparse.py

```diff
@@ -268,6 +667,1250 @@ def _match(self, mask, item):
+    ):
+        if not other:
+            return self.__class__(header_value=instance.header_value)
+
+        if isinstance(other, str):
+            other_header_value = other
+        else:
+            if hasattr(other, 'items'):
+                other = sorted(
+                    other.items(),
+                    key=lambda item: item[1],
+                    reverse=True,
+                )
+            if isinstance(other, (tuple, list)):
+                result = []
+                for element in other:
+                    if isinstance(element, (tuple, list)):
+                        element = _item_qvalue_pair_to_header_element(
+                            pair=element
+                        )
+                    result.append(element)
+                other_header_value = ', '.join(result)
+            else:
+                other_header_value = str(other)
+                if other_header_value == '':
+                    return self.__class__(header_value=instance.header_value)
+
+        if instance.header_value == '':
+            new_header_value = other_header_value
+        else:
+            new_header_value = (
+                (other_header_value + ', ' + instance.header_value)
+                if instance_on_the_right
+                else (instance.header_value + ', ' + other_header_value)
+            )
+
+        return create_accept_language_header(header_value=new_header_value)
+
+
+def create_accept_language_header(header_value):
+    """
+    Create an object representing the ``Accept-Language`` header in a request.
+
+    :param header_value: (``str``) header value
+    :return: If `header_value` is ``None``, an :class:`AcceptLanguageNoHeader`
+             instance.
+
+             | If `header_value` is a valid ``Accept-Language`` header, an
+               :class:`AcceptLanguageValidHeader` instance.
+
+             | If `header_value` is an invalid ``Accept-Language`` header, an
+               :class:`AcceptLanguageInvalidHeader` instance.
+    """
+    if header_value is None:
+        return AcceptLanguageNoHeader()
+    try:
+        return AcceptLanguageValidHeader(header_value=header_value)
+    except ValueError:
+        return AcceptLanguageInvalidHeader(header_value=header_value)
+
+
+def accept_language_property():
+    doc = """
+        Property representing the ``Accept-Language`` header.
+
+        (:rfc:`RFC 7231, section 5.3.5 <7231#section-5.3.5>`)
+
+        The header value in the request environ is parsed and a new object
+        representing the header is created every time we *get* the value of the
+        property. (*set* and *del* change the header value in the request
+        environ, and do not involve parsing.)'
+    """
+
+    ENVIRON_KEY = 'HTTP_ACCEPT_LANGUAGE'
+
+    def fget(request):
+        """
+        Get an object representing the header in the request.
+
+        This creates a new object (and re-parses the header) on every call.
```

Comment:
Remove this last sentence, the parsing shouldn't be that big of a deal. If it is something that ends up dominating a lot of time for users we can figure out how to speed it up at that time. This exposes how it is implemented rather than what it does.